# Configuration Settings for the Rover Rx
septentrio_gnss_driver:
  ros__parameters:
    # GNSS/INS Parameters

    #device: serial:/dev/ttyACM0 
    #device: tcp://10.42.0.127:28784 
    device: serial:/dev/serial/by-id/usb-Septentrio_Septentrio_USB_Device_3641707-if02
    #device: tcp://192.168.3.1:28784

    serial:
      baudrate: 921600
      rx_serial_port: USB1
      hw_flow_control: "off"

    login:
      user: ""
      password: ""

    frame_id: gnss

    aux1_frame_id: aux1

    get_spatial_config_from_tf: true

    use_ros_axis_orientation: true

    receiver_type: gnss

    multi_antenna: true

    datum: Default

    lock_utm_zone: true

    att_offset:
      heading: 0.0
      pitch: 0.0

    ant_type: "Unknown"
    ant_serial_nr: "Unknown"
    ant_aux1_type: "Unknown"
    ant_aux1_serial_nr: "Unknown"

    polling_period:
      pvt: 100
      rest: 500

    use_gnss_time: false

    ins_use_poi: true

    rtk_settings:  
      ntrip_1:
        id: "NTR1"
        caster: "193.178.171.29"
        caster_port: 2101
        username: "ForschArse"
        password: "Lyilfdh4"
        mountpoint: "RTK-3-ETRF"
        version: "v1"
        tls: false
        fingerprint: ""
        rtk_standard: "auto"
        send_gga: "auto"
        keep_open: true
      # ip_server_1:
      #   id: "IPS1"
      #   port: 28784
      #   rtk_standard: "RTCMv3"
      #   send_gga: "auto"
      #   keep_open: true

    publish:
      navsatfix: true
      gpsfix: true
      gpgga: true
      gprmc: true
      gpst: true
      measepoch: true
      pvtcartesian: true
      pvtgeodetic: true
      basevectorcart: true
      basevectorgeod: true
      poscovcartesian: true
      poscovgeodetic: true
      velcovgeodetic: true
      atteuler: true
      attcoveuler: true
      pose: true
      twist: true
      diagnostics: true
      # For GNSS Rx only
      gpgsa: true
      gpgsv: true
      # For INS Rx only
      insnavcart: true
      insnavgeod: true
      extsensormeas: true
      imusetup: true
      velsensorsetup: true
      exteventinsnavcart: true
      exteventinsnavgeod: true
      imu: true
      localization: true
      tf: false

    # logger

    activate_debug_log: false